lib/Optimizer/CodeGen/ConvertToQIRProfile.cpp

```diff
@@ -214,6 +214,15 @@ struct AddFuncAttribute : public OpRewritePattern<LLVM::LLVMFuncOp> {
     bool isAdaptive = convertTo == "qir-adaptive";
     const char *profileName = isAdaptive ? "adaptive_profile" : "base_profile";
 
+    StringRef requiredQubitsStr(std::to_string(info.nQubits));
```

Comment:
This looks like it may be unsafe. `to_string` is returning a new `string` object, but that object isn't interned anywhere so the `StringRef` would be set to reference an object that could be deleted at the end of the statement.

Reply:
> This looks like it may be unsafe. `to_string` is returning a new `string` object, but that object isn't interned anywhere so the `StringRef` would be set to reference an object that could be deleted at the end of the statement.

Yeah, I was surprised this worked. Here is the relevant section of StringRef.h:
```cpp
    /// Disallow accidental assignment from a temporary std::string.
    ///
    /// The declaration here is extra complicated so that `stringRef = {}`
    /// and `stringRef = "abc"` continue to select the move assignment operator.
    template <typename T>
    std::enable_if_t<std::is_same<T, std::string>::value, StringRef> &
    operator=(T &&Str) = delete;
```
So it looks like it would protect me because it would move a temporary if it could, but I should probably update this to be safe.

Reply:
Updated in 96d09dfe3